src/main/java/com/openshift/jenkins/plugins/pipeline/NameValuePair.java

```diff
@@ -0,0 +1,34 @@
+package com.openshift.jenkins.plugins.pipeline;
+
+import hudson.Extension;
+import hudson.model.AbstractDescribableImpl;
+import hudson.model.Descriptor;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+public class NameValuePair extends AbstractDescribableImpl<NameValuePair> {
+
+    protected final String name;
+    protected final String value;
+
+    @DataBoundConstructor
+    public NameValuePair(String name, String value) {
+        this.name = name;
+        this.value = value;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String getValue() {
+        return value;
+    }
+
+    @Extension
+    public static class DescriptorImpl extends Descriptor<NameValuePair> {
+        public String getDisplayName() {
+            return "Name/Value Pairs";
+        }
```

Comment:
Can the user add a name value pair with empty string ... if so, for consistency sake, we could add checks either here or in the DescriptorImpl for OpenShiftBuilder that ensures something is at least set and we don't go down the env var path when we don't need to.